### PR TITLE
Add details regarding JitifyCache

### PR DESCRIPTION
**Pull request by Robadob**
Relates https://github.com/FLAMEGPU/FLAMEGPU2/pull/999

### CODE REVIEW

File: src/guide/agent-functions/defining-agent-functions.rst [hunk 1]
```diff
@@ -87,6 +87,26 @@ Runtime C++ style compiled functions can be used with both the the C++ and Pytho
...
+
+The on-disk cache may grow over time if you are using many different versions of flame gpu and compiling many different agent functions (e.g. running the test suites). If you wish to purge the on-disk cache this can be achieved via the below command.
+
+.. tabs::
+
+  .. code-tab:: cpp C++
+
+    #include "flamegpu/util/detail/JitifyCache.h"
+
+    flamegpu::util::detail::JitifyCache::clearDiskCache()
```
Comment (ptheywood) [inline src/guide/agent-functions/defining-agent-functions.rst:102]: We shouldn't be advertising parts of `detail` for users to call, its not part of the "public" API.

Comment (Robadob) [reply src/guide/agent-functions/defining-agent-functions.rst:102]: How else do you want to expose this then. I have documented it as internal in the above text.

Comment (Robadob) [reply src/guide/agent-functions/defining-agent-functions.rst:102]: This is a case of, they shouldn't **need** it, but it's there if they want.

Comment (Robadob) [reply src/guide/agent-functions/defining-agent-functions.rst:102]: Argument for moving `JitifyCache` out of `detail` and just leaving it in `util`?

Comment (ptheywood) [reply src/guide/agent-functions/defining-agent-functions.rst:102]: Depends on how much of the rest of JitifyCache (and even the name JitifyCache) wants to be fixed in the public API. 

`loadKernel` doesn't seem like something we should be exposing. 
The various `useX`/`clearX` methods maybe? 

Alternatively we could add a new file / static method somewhere, which just calls `JitifyCache::clearDiskCache`. Could just be a `flmamegpu::util` namespace scoped static method like `cleanup()`?

Comment (Robadob) [reply src/guide/agent-functions/defining-agent-functions.rst:102]: Fair, the configuration options of `JitifyCache` are useful but only really for users who don't want defaults and cant configure at cmake time (Python).

I need to do some primage stuff now, so will probably return to this later today/next week.

**Review by ptheywood — APPROVED** [review]

**Pull request merged**